modules/evm/src/runner/handler.rs

```diff
@@ -638,7 +666,12 @@ impl<'vicinity, 'config, 'meter, T: Config> HandlerT for Handler<'vicinity, 'con
 							// try_or_rollback!(self.storage_meter.record_cost(0));
 							TransactionOutcome::Commit(Capture::Exit((s.into(), out)))
 						}
-						Err(e) => TransactionOutcome::Rollback(Capture::Exit((e.into(), Vec::new()))),
+						Err(e) => {
+							let mut w = sp_std::Writer::default();
+							let _ = core::write!(&mut w, "{:?}", e);
+							substate.err_msg = Some(w.into_inner());
+							TransactionOutcome::Rollback(Capture::Exit((e.into(), Vec::new())))
```

Comment:
can we just put the error message here?
```suggestion
							TransactionOutcome::Rollback(Capture::Exit((e.into(), w.into_inner())))
```

Reply:
https://github.com/rust-blockchain/evm/blob/22fe78a32af8a790663ac7cd437da89d1cb54577/runtime/src/eval/system.rs#L404

The code of precompile is empty, so the `out_len = 0`. The error message can't return from precompile.
```
let target_len = min(out_len, U256::from(runtime.return_data_buffer.len()));
```
```
ExitReason::Revert(_) => {
	push_u256!(runtime, U256::zero());

	let _ = runtime.machine.memory_mut().copy_large(
		out_offset,
		U256::zero(),
		target_len,
		&runtime.return_data_buffer[..],
	);

	Control::Continue
}
```